lib/src/main/java/com/basistheory/android/view/CardNumberElement.kt

```diff
@@ -39,8 +43,10 @@ class CardNumberElement @JvmOverloads constructor(
         isEmpty: Boolean,
         isValid: Boolean
     ): ChangeEvent {
-        val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
-        val eventDetails = cardResult.cardDetails?.brand?.let { brand ->
+        val cardMetadata = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
+        this.cardMetadata = cardMetadata
```

Comment:
I think you can delete these lines now, the `cardMetadata` has already been updated by `beforeTextChanged` at this point.

Reply:
my bad, i removed them and then added them to troubleshoot something and forgot to remove them back